packages/image_picker/image_picker_platform_interface/lib/src/method_channel/method_channel_image_picker.dart

```diff
@@ -174,10 +181,21 @@ class MethodChannelImagePicker extends ImagePickerPlatform {
 
     final String? path = result['path'];
 
+    // In this case, multiRetrieve is invoked.
+    if (!isRetrieve) {
+      final pathList = result['pathList'];
+      if (pathList != null) {
+        for (String path in pathList) {
+          pickedFileList.add(PickedFile(path));
+        }
+      }
+    }
+
```

Comment:
The `if (!isRetrieve)` is not really needed. When the old method is used `pathList` is not in the map, but that means that would just return null. 

> The value for the given key, or null if key is not in the map.

https://api.dart.dev/stable/2.13.4/dart-core/Map/operator_get.html